packages/tools/src/types/LabelmapTypes.ts

```diff
@@ -35,7 +35,16 @@ export type LabelmapRenderingConfig = {
   ofun?: vtkPiecewiseFunction;
 };
 
-export type LabelmapSegmentationData = {
+export type LabelmapSegmentationDataVolume = {
   volumeId: string;
   referencedVolumeId?: string;
 };
+
+export type LabelmapSegmentationDataStack = {
+  imageIds: Array<string>;
```

Comment:
Make the referendImageIds a Set, or if you want to get updated imageId, then use a map of referenced to updated imageIds - either way, using a Map will be quite a bit faster for large stacks - you should expect to see stacks in the range of 2000+ images working effectively.